packages/expo/cli/start/server/middleware/resolveAssets.ts

```diff
@@ -0,0 +1,124 @@
+      );
+      // Delete the field so Expo Go doesn't attempt to read it.
+      delete manifest.android.googleServicesFile;
+    }
+  }
+  if (manifest.ios?.googleServicesFile) {
+    try {
+      const contents = await fs.readFile(
+        path.resolve(projectRoot, manifest.ios.googleServicesFile),
+        'base64'
+      );
+      manifest.ios.googleServicesFile = contents;
+    } catch (error) {
+      Log.warn(
+        `Could not parse Expo config: ios.googleServicesFile: "${manifest.ios.googleServicesFile}"`
+      );
+      // Delete the field so Expo Go doesn't attempt to read it.
+      delete manifest.ios.googleServicesFile;
+    }
+  }
+  return manifest;
+}
+
+/**
+ * Get all fields in the manifest that match assets, then filter the ones that aren't set.
+ *
+ * @param manifest
+ * @returns Asset fields that the user has set like ["icon", "splash.image", ...]
+ */
+export async function getAssetFieldPathsForManifestAsync(manifest: ExpoConfig): Promise<string[]> {
+  // String array like ["icon", "notification.icon", "loading.icon", "loading.backgroundImage", "ios.icon", ...]
+  const sdkAssetFieldPaths = await getAssetSchemasAsync(manifest.sdkVersion);
+  return sdkAssetFieldPaths.filter((assetSchema) => get(manifest, assetSchema));
+}
+
+/** Resolve all assets in the app.json inline. */
+export async function resolveManifestAssets(
+  projectRoot: string,
+  {
+    manifest,
+    resolver,
+  }: {
+    manifest: ExpoConfig;
+    resolver: (assetPath: string) => Promise<string>;
+  }
+) {
+  try {
+    // Asset fields that the user has set like ["icon", "splash.image"]
+    const assetSchemas = await getAssetFieldPathsForManifestAsync(manifest);
+    // Get the URLs
+    const urls = await Promise.all(
+      assetSchemas.map(async (manifestField) => {
+        const pathOrURL = get(manifest, manifestField);
+        // URL
+        if (validateUrl(pathOrURL, { requireProtocol: true })) {
+          return pathOrURL;
+        }
+
+        // File path
+        if (await fileExistsAsync(path.resolve(projectRoot, pathOrURL))) {
+          return await resolver(pathOrURL);
+        }
+
+        // Unknown
+        const err: ManifestResolutionError = new CommandError(
+          'MANIFEST_ASSET',
+          'Could not resolve local asset: ' + pathOrURL
+        );
+        err.localAssetPath = pathOrURL;
+        err.manifestField = manifestField;
+        throw err;
+      })
+    );
+
+    // Set the corresponding URL fields
+    assetSchemas.forEach((manifestField, index: number) =>
+      set(manifest, `${manifestField}Url`, urls[index])
+    );
+  } catch (e) {
+    if (e.localAssetPath) {
+      Log.warn(
+        `Unable to resolve asset "${e.localAssetPath}" from "${e.manifestField}" in your app.json or app.config.js`
+      );
+    } else {
+      Log.warn(`Warning: Unable to resolve manifest assets. Icons might not work. ${e.message}.`);
```

Comment:
Fonts seem common too:
```suggestion
      Log.warn(`Warning: Unable to resolve manifest assets. Icons and fonts might not work. ${e.message}.`);
```